app/domain/search/sort.rb

```diff
@@ -4,31 +4,56 @@ class Sort
 
     def self.all
       [
-        Sort.new(:page_views_desc, "six_months_page_views", :desc),
+        Sort.new(:page_views_desc, "Six months page views (Decending)",
+                 "six_months_page_views", :desc),
+        Sort.new(:page_views_asc, "Six months page views (Ascending)",
+                 "six_months_page_views", :asc),
+        Sort.new(:title_desc, "Title (Decending)",
+                 "title", :desc),
+        Sort.new(:title_asc, "Title (Ascending)",
+                 "title", :asc),
+        Sort.new(:public_updated_at_asc, "Last public update (Oldest first)",
+                 "public_updated_at", :asc, nulls: :first),
+        Sort.new(:public_updated_at_desc, "Last public update (Most recent first)",
+                 "public_updated_at", :desc, nulls: :last),
       ]
     end
 
-    attr_reader :identifier
+    attr_reader :identifier, :label
 
-    def initialize(identifier, field, order)
+    def initialize(identifier, label, field, order, options = {})
       @identifier = identifier
+      @label = label
       @field = field
       @order = order
+      @nulls = options.fetch(:nulls, order == :dsc ? :first : :last)
     end
 
     def apply(scope)
-      scope.order("#{field} #{order}")
+      scope.order("#{field} #{order} NULLS #{nulls}")
     end
 
     def where_only_after(scope, content_item)
-      scope.where(
-        "#{field} #{{ desc: '<=', asc: '>=' }[order]} ?",
-        content_item.object.send(field)
-      )
+      field_value = content_item.object.send(field)
+
+      if field_value.nil?
+        case @nulls
+        when :first
+          # Can't narrow down the possibilities any more, as the field
+          # value for the next item could be null, or it might not be.
+          scope
+        when :last
+          scope.where("#{field} IS NULL")
+        end
+      else
+        scope.where(
+          "#{field} #{{ desc: '<=', asc: '>=' }[order]} ?", field_value
+        )
+      end
```

Comment:
The `where_only_after` function takes a scope and content_item, and returns a new scope based on the one provided, but where the scope has been narrowed as much as possible to only include items after the provided item.

From the Sort class, you know the field that is being used for the sort, and the order. From the provided content item, you know the value of the field which is being used in the sort. From these three bits of information, you can generate a new condition to apply to the scope to further limit it to only items which have a sort value equal to or after (this doesn't quite match the naming, it needs weakening).

This approach won't work where the value of the field that the sort is using is nil, as that will be `NULL` in the query. However, the sort includes the information on where the null values are placed. If they are first, then I don't think there is a way of narrowing down the possibilities, so just return the scope. If the null values are last, you can at least eliminate anything that is not null, so do that.

Does that make some amount of sense?